src/promptflow-tools/promptflow/tools/aoai.py

```diff
@@ -156,6 +176,9 @@ def embedding(self, input, deployment_name: str, user: str = ""):
             headers={"ms-azure-ai-promptflow-called-from": "aoai-tool"},
             **self._connection_dict,
         )
+        if RecordStorage.is_recording_mode():
```

Comment:
can this also be moved to decorator function?

Reply:
Okay, let's make it.